baggage/views.py

```diff
@@ -0,0 +1,223 @@
+    template_name = 'baggage_list.html'
+    table_class = BaggageListTable
+    filterset_class = BaggageListFilter
+    table_pagination = {'per_page': 100}
+
+    def get_current_tabs(self):
+        return organizer_tabs(self.request.user)
+
+    def get_queryset(self):
+        if 'user_id' in self.kwargs:
+            id_ = self.kwargs['user_id']
+            user = User.objects.filter(id=id_)
+            return Bag.objects.filter(status=BAG_ADDED, owner=user)
+        return Bag.objects.filter(status=BAG_ADDED)
+
+
+class BaggageHacker(IsVolunteerMixin, TabsViewMixin, SingleTableMixin, FilterView):
+    template_name = 'baggage_hacker.html'
+    table_class = BaggageListTable
+    filterset_class = BaggageListFilter
+    table_pagination = {'per_page': 100}
+
+    def get_back_url(self):
+        return 'javascript:history.back()'
+
+    def get_queryset(self):
+        id_ = self.kwargs['user_id']
+        user = User.objects.filter(id=id_)
+        return Bag.objects.filter(status=BAG_ADDED, owner=user)
+
+
+class BaggageUsers(IsVolunteerMixin, TabsViewMixin, SingleTableMixin, FilterView):
+    template_name = 'baggage_users.html'
+    table_class = BaggageUsersTable
+    filterset_class = BaggageUsersFilter
+    table_pagination = {'per_page': 100}
+
+    def get_current_tabs(self):
+        return organizer_tabs(self.request.user)
+
+    def get_queryset(self):
+        return CheckIn.objects.all()
+
+
+class BaggageAdd(IsVolunteerMixin, TabsView):
+    template_name = 'baggage_add.html'
+
+    def get_back_url(self):
+        return 'javascript:history.back()'
+
+    def get_context_data(self, **kwargs):
+        context = super(BaggageAdd, self).get_context_data(**kwargs)
+        userid = kwargs['new_id']
+        user = User.objects.filter(id=userid).first()
+        if not user:
+            raise Http404
+        context.update({
+            'user': user,
+            'rooms': Room.objects.all()
+        })
+        return context
+
+    def post(self, request, *args, **kwargs):
+        bagtype = request.POST.get('bag_type')
+        bagcolor = request.POST.get('bag_color')
+        bagdesc = request.POST.get('bag_description')
+        bagspe = request.POST.get('bag_special')
+        userid = request.POST.get('user_id')
+        bagimage = request.POST.get('bag_image')
+
+        bag = Bag()
+        bag.owner = User.objects.filter(id=userid).first()
+        bag.inby = request.user
+        bag.btype = bagtype
+        bag.color = bagcolor
+        bag.description = bagdesc
+        bag.special = (bagspe == 'special')
+
+        if bagimage:
+            try:
+                bagimageformat, bagimagefile = bagimage.split(';base64,')
+                bagimageext = bagimageformat.split('/')[-1]
+                bag.image = ContentFile(base64.b64decode(bagimagefile),
+                                        name=(str(time.time()).split('.')[0] + '-' + userid + '.' + bagimageext))
+            except:
+                pass
```

Comment:
Consider adding at least a log to know something went wrong

Reply:
Print added!